src/pyodide/internal/tarfs.ts

```diff
@@ -18,19 +17,22 @@ const FSOps = {
     }
   },
   readdir(node) {
-    return Array.from(node.info.children.keys());
+    return Array.from(node.info.children!.keys());
   },
   lookup(parent, name) {
-    return parent.info.children.get(name);
+    return parent.info.children!.get(name)!;
   },
   read(stream, position, buffer) {
-    return stream.node.info.reader.read(
+    if (stream.node.contentsOffset == undefined) {
+      throw new Error("contentsOffset is undefined");
+    }
```

Comment:
Can we change the types so that these fields are guaranteed to not be undefined?

Reply:
contentsOffset is undefined when the node is a directory 😶